drivers/flash/Kconfig.simulator

```diff
@@ -0,0 +1,158 @@
+
+config FLASH_SIMULATOR_SIMULATE_FAILURES
+	bool
+	prompt "Enable failure simulation"
+	help
+	 This setting enables simulation of various API and hardware failures,
+	 which can be used for testing.
+
+config FLASH_SIMULATOR_MAX_ERASE_CYCLES
+	int
+	prompt "Maximum number of erase cycles"
+	default 10000
+	depends on FLASH_SIMULATOR_SIMULATE_FAILURES
+	help
+	 Maximum number of erase cycles after which the flash can
+	 no longer be reliably erased.
+
+config FLASH_SIMULATOR_READ_API_FAILURE_RATE_PM
+	int
+	prompt "Read API failure rate ‰ (per mille)"
+	default 0
+	range 0 1000
+	depends on FLASH_SIMULATOR_SIMULATE_FAILURES
+	help
+	 This setting controls the read API failure rate in units of 1/1000.
+	 This setting is intended for testing.
+
+config FLASH_SIMULATOR_WRITE_API_FAILURE_RATE_PM
+	int
+	prompt "Write API failure rate ‰ (per mille)"
+	default 0
+	range 0 1000
+	depends on FLASH_SIMULATOR_SIMULATE_FAILURES
+	help
+	 This setting controls the write API failure rate in units of 1/1000.
+	 This setting is intended for testing.
+
+config FLASH_SIMULATOR_ERASE_API_FAILURE_RATE_PM
+	int
+	prompt "Erase API failure rate ‰ (per mille)"
+	default 0
+	range 0 1000
+	depends on FLASH_SIMULATOR_SIMULATE_FAILURES
+	help
+	 This setting controls the erase API failure rate in units of 1/1000.
+	 This setting is intended for testing.
+
+config FLASH_SIMULATOR_READ_HW_FAILURE_RATE_PM
+	int
+	prompt "Hardware read failure rate ‰ (per mille)"
+	default 0
+	range 0 1000
+	depends on FLASH_SIMULATOR_SIMULATE_FAILURES
+	help
+	 This setting controls the hardware failure rate during read operations in units of 1/1000.
+	 This setting is intended for testing.
+
+ config FLASH_SIMULATOR_WRITE_HW_FAILURE_RATE_PM
+	int
+	prompt "Hardware write failure rate ‰ (per mille)"
+	default 0
+	range 0 1000
+	depends on FLASH_SIMULATOR_SIMULATE_FAILURES
+	help
+	 This setting controls the hardware failure rate during write operations in units of 1/1000.
+	 This setting is intended for testing.
+
+	config FLASH_SIMULATOR_ERASE_HW_FAILURE_RATE_PM
+	int
+	prompt "Hardware erase failure rate ‰ (per mille)"
+	default 0
+	range 0 1000
+	depends on FLASH_SIMULATOR_SIMULATE_FAILURES
+	help
+	 This setting controls the hardware failure rate during erase operations in units of 1/1000.
+	 This setting is intended for testing.
+
+config FLASH_SIMULATOR_SIMULATE_TIMING
+	bool
+	prompt "Enable hardware timing simulation"
+
+config FLASH_SIMULATOR_MIN_READ_TIME_US
+	int
+	prompt "Minimum read time (µS)"
+	depends on FLASH_SIMULATOR_SIMULATE_TIMING
+	default 2
+	range 1 1000000
+
+config FLASH_SIMULATOR_MIN_WRITE_TIME_US
+	int
+	prompt "Minimum write time (µS)"
+	depends on FLASH_SIMULATOR_SIMULATE_TIMING
+	default 100
+	range 1 1000000
+
+config FLASH_SIMULATOR_MIN_ERASE_TIME_US
+	int
+	prompt "Minimum erase time (µS)"
+	depends on FLASH_SIMULATOR_SIMULATE_TIMING
+	default 2000
+	range 1 1000000
+
+config FLASH_SIMULATOR_ERASE_PROTECT
+	bool
```

Comment:
`bool y` - should be enabled by default.

Reply:
Agreed